platform/security/core/security-core-impl/src/main/java/ddf/security/common/util/SecurityUtils.java

```diff
@@ -0,0 +1,217 @@
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package ddf.security.common.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.AccessController;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Dictionary;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.karaf.jaas.boot.principal.RolePrincipal;
+import org.codice.ddf.security.handler.api.PKIAuthenticationToken;
+import org.codice.ddf.security.handler.api.PKIAuthenticationTokenFactory;
+import org.codice.ddf.security.handler.api.UPAuthenticationToken;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceReference;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.security.Subject;
+import ddf.security.assertion.SecurityAssertion;
+import ddf.security.service.SecurityManager;
+import ddf.security.service.SecurityServiceException;
+
+public class SecurityUtils {
+
+    private static final RolePrincipal ADMIN_ROLE = new RolePrincipal("admin");
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SecurityUtils.class);
+
+    private static Subject cachedSystemSubject;
+
+    private static Object lock = new Object();
+
+    public SecurityUtils() {
+
+    }
+
+    public Subject getSubject(String username, String password) {
+        UPAuthenticationToken token = new UPAuthenticationToken(username, password);
+        SecurityManager securityManager = getSecurityManager();
+        if (securityManager != null) {
+            try {
+                return securityManager.getSubject(token);
+            } catch (Exception sse) {
+                LOGGER.error("Unable to request subject for {} user.", username, sse);
+            }
+        }
+        return null;
+    }
+
+    public boolean tokenAboutToExpire(Subject subject) {
+        boolean tokenAboutToExpire = true;
+        if ((null != subject) && (null != subject.getPrincipals()) && (null != subject
```

Comment:
Done. I also when back to the AnonymousInterceptor where I copied this code from and had it use the Security.tokenAboutToExpire so we don't have code duplication.